src/main/java/org/springframework/data/elasticsearch/core/SearchOperations.java

```diff
@@ -170,32 +202,203 @@ default long count(Query query, IndexCoordinates index) {
+		return (ScrolledPage<T>) SearchHitSupport
+				.unwrapSearchHits(searchScrollContinue(scrollId, scrollTimeInMillis, clazz));
+	}
 
-	<T> ScrolledPage<T> continueScroll(@Nullable String scrollId, long scrollTimeInMillis, Class<T> clazz);
+	/**
+	 * Returns next scrolled page
+	 *
+	 * @param scrollId the scroll id
+	 * @param scrollTimeInMillis duration of the scroll time
+	 * @param clazz The class of entity to retrieve.
+	 * @return scrolled page result
+	 */
+	<T> ScrolledPage<SearchHit<T>> searchScrollContinue(@Nullable String scrollId, long scrollTimeInMillis,
+			Class<T> clazz);
 
 	/**
 	 * Clears the search contexts associated with specified scroll ids.
 	 *
 	 * @param scrollId the scroll id
+	 * @deprecated since 4.0, use {@link #searchScrollClear(String)}.
 	 */
-	void clearScroll(String scrollId);
+	@Deprecated
+	default void clearScroll(String scrollId) {
+		searchScrollClear(scrollId);
+	}
 
 	/**
 	 * more like this query to search for documents that are "like" a specific document.
 	 *
+	 * @param <T> element return type
+	 * @param query the query to execute
+	 * @param clazz the entity clazz used for property mapping
+	 * @param index the index to run the query against
+	 * @return page with the results
+	 * @deprecated since 4.0, use {@link #searchMoreLikeThis(MoreLikeThisQuery, Class, IndexCoordinates)}.
+	 */
+	@Deprecated
+	default <T> AggregatedPage<T> moreLikeThis(MoreLikeThisQuery query, Class<T> clazz, IndexCoordinates index) {
+		return (AggregatedPage<T>) SearchHitSupport.unwrapSearchHits(searchMoreLikeThis(query, clazz, index));
+	}
+
+	/**
+	 * Execute the query against elasticsearch and return the first returned object.
+	 *
+	 * @param query the query to execute
+	 * @param clazz the entity clazz used for property mapping
+	 * @param index the index to run the query against
+	 * @return the first matching object
+	 */
+	default <T> SearchHit<T> searchForObject(Query query, Class<T> clazz, IndexCoordinates index) {
+		List<SearchHit<T>> content = searchForPage(query, clazz, index).getContent();
+		return content.isEmpty() ? null : content.get(0);
+	}
+
+	/**
+	 * Execute the query against elasticsearch and return result as {@link Page}.
+	 *
+	 * @param query the query to execute
+	 * @param clazz the entity clazz used for property mapping
+	 * @param index the index to run the query against
+	 * @return a page with aggregations
+	 */
+	<T> AggregatedPage<SearchHit<T>> searchForPage(Query query, Class<T> clazz, IndexCoordinates index);
+
+	/**
+	 * Execute the multi-search against elasticsearch and return result as {@link List} of {@link Page}
+	 *
+	 * @param queries the queries
+	 * @param clazz the entity clazz used for property mapping
+	 * @param index the index to run the query against
+	 * @return list of pages with the results
+	 */
+	<T> List<Page<SearchHit<T>>> searchForPage(List<? extends Query> queries, Class<T> clazz, IndexCoordinates index);
```

Comment:
Let's rename this method to `multiSearch`. I also wonder what's the benefit of having `List<Page<…>>` over `List<SearchHits<T>>` in the return type.

Reply:
The only need of having the page returned instead of the list is that this currently is the only possibility to get aggregations returned. Aggregations don't belong in the page but in the `SearchHits`. I have created a ticket for moving them there, so when we have that implemented, we can consolidate the different methods returning `List` and `Page` and remove the page variants because the page returned does not hold relevant info then. This affects not only _multiSearch_  but the single query _search_ as well.

Reply:
I renamed it to `multiSearchForPage` and changed the overload that retruns a `List<SearchHits<T>>` to `multiSearch`. We should aim to keep the methods returning the `SearchHits` and not the ones with the `AggregatedPage`s once we have the necessary information in the `SearchHits`class.

Reply:
Thanks for the detail. I also felt that aggregations should go into `SearchHits` (or a subtype of it).